teps/0119-add-taskrunspec-template-in-pipelinerun.md

```diff
@@ -0,0 +1,289 @@
+---
+status: proposed
+title: Add taskRunSpec template in PipelineRun
+creation-date: '2022-08-12'
+last-updated: '2022-08-12'
+authors:
+- '@yuzp1996'
+---
+
+# TEP-0119: Add taskRunSpec template in PipelineRun
+
+<!-- toc -->
+- [Summary](#summary)
+  - [Use Cases](#use-cases)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+  - [Former implement](#former-implement)
+  - [implement](#implement)
+  - [Test Plan](#test-plan)
+<!-- /toc -->
+
+## Summary
+User is supported to configure specify specific taskrun spec when they create pipelinerun. However, users is not supported to specify spec for all taskrun. 
+If you want to specify a common configuration for all taskrun they must specify them individually.
+
+This Tep is try to provide a convenient way that specify all taskrun spec in pipelinerun.
+
+### Use Cases
+Pipeline and Task User:
+I would like to use the same service account for all of my pipeline tasks except one.
+I want to specify compute resources that each TaskRun in my PipelineRun should run with, and don't want to have to specify them individually for each TaskRun.
+
+## Proposal
+Add filed taskRunSpecTemplate to PipelineRun.Spec then we can specify the common configuration in taskRunSpecTemplate, and the configuration will apply to all the TasRun.
+
+```yaml
```

Comment:
It would be good to have an example of how the new field would be used to meet one of these use cases. For example, for use case number 1 (service accounts), the user might have a pipeline that clones and then runs unit tests, integration tests, and linters. They might want to use the same service account for all the tests, but a different service account for the clone. It would be helpful to include the pipeline definition as well. In this example, the taskRunTemplate has the same configuration as taskRunSpecs, which a user is unlikely to do.